server/src/test/java/com/objectcomputing/checkins/services/pulseResponse/PulseResponseControllerTest.java

```diff
@@ -0,0 +1,256 @@
+
+        final HttpResponse<?> response = client.toBlocking().exchange(HttpRequest.GET(String.format("/?dateFrom=%tF&dateTo=%tF", testDateFrom, testDateTo)));
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertEquals(2, response.getContentLength());
+    }
+
+    // Find By TeamMemberId
+    @Test
+    public void testGetFindByTeamMemberId() {
+
+        setupTestData();
+
+        final HttpResponse<?> response = client.toBlocking().exchange(HttpRequest.GET(String.format("/?teamMemberId=%s", testTeamMemberId)));
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertNotEquals(2, response.getContentLength());
+    }
+ 
+    // Find By findBySubmissionDateBetween
+    @Test
+    public void testGetFindByfindBySubmissionDateBetween() {
+
+        setupTestData();
+
+        LocalDate testDateFrom = LocalDate.of(2019, 1, 01);
+        LocalDate testDateTo = LocalDate.of(2021, 1, 01);
+
+        final HttpResponse<?> response = client.toBlocking().exchange(HttpRequest.GET(String.format("/?dateFrom=%tF&dateTo=%tF", testDateFrom, testDateTo)));
+        
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertNotEquals(2, response.getContentLength());
+    }
+
+    // POST - Valid Body
+    @Test
+    public void testPostSave() {
+
+        final Map<String, Object> testBody = new HashMap<String, Object>() {{
+            put("submissionDate", testsubmissionDate);
+            put("updatedDate", testUpdatedDate);
+            put("teamMemberId", testTeamMemberId);
+            put("internalFeelings", "operation");
+            put("externalFeelings", "successful");
+        }};
+
+        final HttpResponse<PulseResponse> response = client.toBlocking().exchange(HttpRequest.POST("", testBody));
+        assertEquals(HttpStatus.CREATED, response.getStatus());
+        assertNotNull(response.getContentLength());
+    }
+
+    // PUT - Valid Body
+    @Test
+    public void testPutUpdate() {
+
+        UUID testId = UUID.randomUUID();
+        final Map<String, Object> testBody = new HashMap<String, Object>() {{
+            put("id", testId);
+            put("submissionDate", testsubmissionDate);
+            put("updatedDate", testUpdatedDate);
+            put("teamMemberId", testTeamMemberId);
+            put("internalFeelings", "operation");
+            put("externalFeelings", "successful");
+        }};
+
+        PulseResponse testPulseResponse = new PulseResponse(testsubmissionDate, testUpdatedDate, testTeamMemberId, "Internal Feeling", "External Feeling");
+        testPulseResponse.setId(testId);
+
+        when(mockPulseResponseRepository.update(testPulseResponse)).thenReturn(testPulseResponse);
+
+        final HttpResponse<?> response = client.toBlocking().exchange(HttpRequest.PUT("", testBody));
+        assertEquals(HttpStatus.OK, response.getStatus());
+        assertNotNull(response.getContentLength());
+
+    }
```

Comment:
I notice there's not a test for updating a non-existent PulseResponse (one where the ID provided doesn't exist/isn't correct). Further, looking at your service layer code, I don't think that it is likely handled gracefully. I would consider adding a test for that and ensuring that your service responds accordingly.

Reply:
Good catch! I've added the validation for this and have moved all logic to a service class.

if there's no ID in HttpRequest -> throws Bad Request
if ID doesn't exist in DB -> throws Not Found
If ID is valid -> updates record in DB